### PR TITLE
Feat: support new ansible template type

### LINKED ISSUE #926: Support new `ansible` template type
**Is your feature request related to a problem? Please describe.**
We added a new template type - `ansible` :)

**Describe the solution you'd like**
Add it as an option to deploy (exactly like all other types)

**some key points:**
1. We support version 3.0.0 and above
2. it has dedicated steps in custom flows :
    a. ansibleGalaxy
    b. ansibleCheck
    c. ansiblePlaybook

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #926 

### Solution

1. Updated the template API structs.
2. Updated the schema and implemented support.
3. Updated acceptance tests.
4. Added a harness test.
5. Some other misc changes (part of linting/bug fixes effort).

### CODE REVIEW

File: env0/data_template.go [hunk 1]
```diff
@@ -49,7 +47,7 @@ func dataTemplate() *schema.Resource {
 			},
 			"type": {
 				Type:        schema.TypeString,
-				Description: fmt.Sprintf("template type (allowed values: %s)", strings.Join(allowedTemplateTypes, ", ")),
+				Description: "the template type",
```
Comment (Wassap124) [inline env0/data_template.go:50]: why not show all available types? 🤔

**Review by Wassap124 — COMMENTED** [review]
Sorry, I missed that we also accept `latest` version 🙏

**Comment by Wassap124** [comment]
[Support Ansible in terraform provider](https://trello.com/c/AHwJFV4Y/13872-support-ansible-in-terraform-provider)

File: env0/data_template.go (hunk 1 above)
Comment (TomerHeber) [reply env0/data_template.go:50]: this is a computed field.  I can add it. But it serves no purpose.
In other words, this is not an input field, but an output field.
If you do think it serves some purpose, I can bring it back.
(In the resource schema - where it's an input, it's still there).

File: client/template.go [hunk 2]
```diff
@@ -122,32 +127,51 @@ func (payload *TemplateCreatePayload) Invalidate() error {
...
+
+		c, _ := semver.NewConstraint(">= 3.0.0")
+
+		v, err := semver.NewVersion(payload.AnsibleVersion)
+		if err != nil {
+			return fmt.Errorf("invalid ansible version '%s': %w", payload.AnsibleVersion, err)
+		}
+
+		if !c.Check(v) {
+			return errors.New("supported ansible versions are 3.0.0 and above")
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply client/template.go:174]: I have added latest, and updated the tests as well.

File: env0/resource_template.go [hunk 3]
```diff
@@ -255,6 +256,13 @@ func getTemplateSchema(prefix string) map[string]*schema.Schema {
 			Optional:    true,
 			Default:     false,
 		},
+		"ansible_version": {
+			Type:             schema.TypeString,
+			Description:      "the ansible version to use (required when the template type is 'ansible'). Supported versions are 3.0.0 and above",
+			Optional:         true,
+			ValidateDiagFunc: NewRegexValidator(`^(?:[0-9]{1,2}\.[0-9]{1,2}\.[0-9]{1,2})|$`),
+			Default:          "",
+		},
```
[the remark this answers is not in the document]
Comment (TomerHeber) [reply env0/resource_template.go]: added,

File: env0/data_template.go (hunk 1 above)
Comment (Wassap124) [reply env0/data_template.go:50]: no no, i thought it was input, that sounds fine

**Review by Wassap124 — APPROVED** [review]
great work 👍

**Pull request merged**